index.bs

```diff
@@ -895,47 +895,49 @@ spec: webidl
       </dt>
       <dd>
         A list of {{MediaDeviceInfo/deviceId}} values for the devices the user
-        has granted access to.
-      </dd>
-      <dt>
-        <a>permission result type</a>
-      </dt>
-      <dd>
-        TODO
+        has made a non-default decision on access to.
       </dd>
       <dt>
         <a>permission query algorithm</a>
       </dt>
       <dd>
-        TODO
+        The permission query algorithm runs the following steps:
+        <ol class="algorithm">
+        <li>
+        If |permissionDesc|.deviceId exists in the <a>extra permission data</a>,
+        set <code><var>status</var>.state</code>
+        to |permissionDesc|'s <a>permission state</a> and terminate these
+        steps.
+        </li>
+        <li>Set <var>global</var> to the {{PermissionDesc}} for the descriptor
+        that results from removing the "deviceId" value.
+        </li>
+        <li>
+        Set <code><var>status</var>.state</code> to <var>global</var>'s
```

Comment:
Do you want to have a way for the site to request the list of all granted `deviceId`s? If so, you'll need to extend the permission result type too, with a list of deviceIds and/or MediaStreams.